sklego/meta/grouped_transformer.py

```diff
@@ -1,9 +1,10 @@
+import narwhals as nw
```

Comment:
@MarcoGorelli I reframed this to rely solely on narwhals.
There are a few missing bits still:

- in `_grouped_utils` there are some useful checks (again on datatypes)
- ln 110 is somewhat problematic in the tests because it is passing a native series without converting, but it would require some more logic for the None case (kinda related to [narwhals#144](https://github.com/narwhals-dev/narwhals/pull/144))

Finally, `__transform_groups` method, I am not sure there is an alternative way without using some sort of index trick, due to the fact that the original ordering needs to be maintained.

Let me know what you think, but I would say that we are quite close to a solution

Reply:
wow, amazing! exciting stuff

I'll add the equivalent of `X_group.select_dtypes(include="number")` and `X_group.select_dtypes(exclude="number")` then, anything else in Narwhals you'd like prioritising please let me know

Reply:
@MarcoGorelli matter of fact, I don't see why we only check for numbers. I solved this as:

```py
X.select(nw.col(groups).is_null().any()).to_numpy().squeeze().any()
```